assimilation_code/modules/utilities/utilities_mod.f90

```diff
@@ -480,6 +481,111 @@ subroutine check_namelist_read(iunit, iostat_in, nml_name)
 
 end subroutine check_namelist_read
 
+
+!-----------------------------------------------------------------------
+!> convert integers or strings describing options into integer. 
+!> 
+!> Some namelist items have historically been integers to describe inflation
+!> algorithms or ... which is a terrible thing to do to the users.
+!> This routine allows for backward compatibility and still provides
+!> the option to specify   inf_flavor = 'VARYING_SS_INFLATION' for example.
+
```

Comment:
this is a routine in the utilities module, so it needs to be pretty general.  i know it was first implemented for inflation, but it needs to clearly be documented as working for anything.

```suggestion
!> Some namelist items have historically been integers which do not help describe the
!> option they are selecting.  They are being converted to descriptive strings.
!> During a transition period before the integers are deprecated this routine will return 
!> an integer value if the input string is either an integer or a string.  
!> For strings, the corresponding integer values are input as a 1-to-1 array with the
!> valid strings.   On error this routine prints an error message and does not return.
!> If needed, this routine could be changed to take an optional return code, which if 
!> present would return to the calling code without printing or calling the error 
!> handler so the caller can take an alternative code path.

```